src/services/expenses.service.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+let EXPENSES = [];
+
+const getExpenses = () => {
+  return EXPENSES;
+};
+
+const getExpenseById = (id) => {
+  return EXPENSES.find((e) => e.id === +id) || null;
+};
+
+const getExpensesByQuery = (query) => {
+  return [...EXPENSES].filter((expense) => {
+    for (const key in query) {
+      switch (key) {
+        case 'id':
+        case 'userId':
+          if (expense[key] !== +query[key]) {
+            return false;
+          }
+          break;
+        case 'from':
+        case 'to':
+          if (key === 'from' && expense.spentAt < query.from) {
+            return false;
+          }
+
+          if (key === 'to' && expense.spentAt > query.to) {
+            return false;
+          }
+          break;
+        case 'categories':
+          if (expense.category !== query.categories) {
```

Comment:
is this check work properly?

Reply:
yep, I just aimed to minimize the size of this function